src/api/java/com/minecolonies/api/research/interfaces/IResearchEffect.java

```diff
@@ -0,0 +1,53 @@
+package com.minecolonies.api.research.interfaces;
+
+import net.minecraft.util.text.TranslationTextComponent;
+
+/**
+ * The effect of a research.
+ * @param <T> the type of the value.
+ */
+public interface IResearchEffect<T>
```

Comment:
This is named wrongly, or implemented wrongly.

As it states right now this is implemented as research effect type. However an effect type should have no notion of its branch, description etc.

Especially should an effect have no notion of its parent..... An effect should be active or not, based on if a research marks it as active or not....

This should be redesigned, cause right now it is a bit of a mess.